google-cloud-contrib/google-cloud-logging-logback/src/main/java/com/google/cloud/logging/contrib/LoggingAppender.java

```diff
@@ -0,0 +1,266 @@
+ * <li>&lt;level&gt;ERROR&lt;/level&gt; (Optional, defaults to "INFO" : logs at or above this level)</li>
+ * <li>&lt;flushLevel&gt;WARNING&lt;/flushLevel&gt; (Optional, defaults to "ERROR")</li>
+ * <li>&lt;resourceType&gt;&lt;/resourceType&gt; (Optional, auto detects on App Engine Flex, Standard, GCE and
+ * GKE, defaults to "global".
+ * See <a href="https://cloud.google.com/logging/docs/api/v2/resource-list">supported resource
+ * types</a></li>
+ * <li>&lt;enhancers&gt;com.example.enhancer1, com.example.enhancer2&lt;/enhancers&gt;</li>
+ * <li>(Optional) add custom labels to log entries using {@link LoggingEnhancer} classes.</li>
+ * <li>&lt;/appender&gt;</li>
+ * </ul>
+ */
+public class LoggingAppender extends UnsynchronizedAppenderBase<ILoggingEvent> {
+
+  private static final String LEVEL_NAME_KEY = "levelName";
+  private static final String LEVEL_VALUE_KEY = "levelValue";
+
+  private volatile Logging logging;
+  private List<LoggingEnhancer> loggingEnhancers;
+  private WriteOption[] defaultWriteOptions;
+
+  private Level flushLevel;
+  private String log;
+  private Level logLevel;
+  private String resourceType;
+  private String enhancers;
+
+  /**
+   * Batched logging requests get immediately flushed for logs at or above level.
+   *
+   * <p>Defaults to Error if not set.
+   *
+   * @param flushLevel Logback log level
+   */
+  public void setFlushLevel(Level flushLevel) {
+    this.flushLevel = flushLevel;
+  }
+
+  /**
+   * Sets the log filename, defaults to java.log.
+   *
+   * @param log filename
+   */
+  public void setLog(String log) {
+    this.log = log;
+  }
+
+  /**
+   * Sets the minimum logging level, defaults to INFO
+   *
+   * @param level logback level
+   */
+  public void setLevel(Level level) {
+    this.logLevel = level;
+  }
+
+  /**
+   * Sets the name of the monitored resource (Optional).
+   *
+   * <p>Must be a <a href="https://cloud.google.com/logging/docs/api/v2/resource-list">supported</a>
+   * resource type. gae_app, gce_instance and container are auto-detected, defaults to "global"
+   *
+   * @param resourceType name of monitored resource
+   */
+  public void setResourceType(String resourceType) {
+    this.resourceType = resourceType;
+  }
+
+  /**
+   * Add extra labels using classes that implement {@link LoggingEnhancer}.
+   *
+   * @param enhancers list of enhancer classnames
+   */
+  public void setEnhancers(String enhancers) {
+    this.enhancers = enhancers;
+  }
+
+  Level getFlushLevel() {
+    return (flushLevel != null) ? flushLevel : Level.ERROR;
+  }
+
+  String getLogName() {
+    return (log != null) ? log : "java.log";
+  }
+
+  Level getLogLevel() {
+    return (logLevel != null) ? logLevel : Level.INFO;
+  }
+
+  MonitoredResource getMonitoredResource(String projectId) {
+    return MonitoredResourceUtil.getResource(projectId, resourceType);
+  }
+
+  List<LoggingEnhancer> getEnhancers() {
```

Comment:
There is a moderate amount of code duplication with JUL LoggingHandler for setting up the enhancers.  Perhaps a bit more of it could be moved to common code?